internal/controller/binding/binding.go

```diff
@@ -150,7 +150,16 @@ func (c *external) Observe(ctx context.Context, mg resource.Managed) (managed.Ex
 	bindingFound := false
 	isResourceLateInitialized := false
 	isBindingUptoDate := false
-	binding := getBinding(bindings, cr)
+	var binding *rabbithole.BindingInfo
+
+	if cr.Status.AtProvider.Vhost != "" {
```

Comment:
What happen if I create 2 identical bindings with the parameters but different `metadata.name`?

So I'm creating one biding with name A set all the parameters and routing key `#`. Then I'm creating binding with name B but different routing key. Will then both override each other? Because from my perspective we should instead of vhost at status we should use a name and in case of that kind of conflicts return errors during reconciliation process with the info that for example given routing key has a conflict with different binding (+ name) and update only the existing routing it was managed under the same name. Maybe we should have some diagram how it behaves in different scenarios?

Reply:
Maybe we should use properties key (I see API has that kind of object) with the info about the k8s object which is responsible for that biding - otherwise I affraid we end up with some lack of integrity between k8s object and rabbit binding (no identification which k8s object creates which binding). Did you check how it's handled in terraform?